Bookstore jpa liquibase

azzilian opened the pull request:
Added - update and delete methods
UpdateBookRequestDto create for updating methods

azzilian committed:
Pom - added dependencies for springboot,mysql,lombok
application.properties - added properties to connect to DB
OnlineBookStoreApplication - added simple test
Book - entity added
BookService + BookRepository - methods save + finAll ad

azzilian committed:
BookRepository - empty lines removed
BookRepositoryImpl - proper exception used now
Book - removed annotations for size of columns
application.properties - l & p removed

azzilian committed:
test application.properties - l & p removed

azzilian committed:
BookDto - created as a response DTO
CreateBookRequestDto - as a request DRO
pom - added dependencies & plugins & properties for mapstruck & lombok
BookController - added methods getAll(), getBookById, createBook
Updated service & repository with new method
New exception added - EntityNotFoundException
Bean removed from OnlineBookStoreApplication
application.properties - added jackson.deserialization
BookMapper added - using MapStruct to format toDto and toModel

azzilian committed:
Resolve merge conflict by incorporating both suggestions

azzilian committed:
Data - remove @Data used getter and setter instead
Exception was lost during merge - returned

azzilian committed:
Updated getbyid - findbyid - now ise optional to process null

azzilian committed:
minor fixes

azzilian committed:
Removed @Response from EntityNotFoundException
Renamed method findBOOKById -> findById

azzilian committed:
removed id from CreateBookRequestDto

azzilian committed:
minor fixes

azzilian committed:
changed EntityNotFound -> DataProcessingException

azzilian committed:
Merge remote-tracking branch 'origin/master' into bookstore-JPA-Liquibase

azzilian committed:
Added - update and delete methods
UpdateBookRequestDto create for updating methods

azzilian committed:
minor fixes

azzilian committed:
soft delete added

azzilian committed:
Luiquibase - implemented with master and create table

karma-o reviewed (changes requested):
Looks good so far, just take a look at the comments

azzilian committed:
Fixed - columns for secrption and iimageurk - now its type Text
Added - liquibase.properties to be able to do rollbacks
Changed depricated @Where to @SqlRestrictions
EntityNotFoundException - reverted

azzilian committed:
UpdateBookRequestDto.java removed and CreateBookRequestDto.java changed to BookRequestDto - changed in code
Added 201 status for post method

andrii-hoienko reviewed (approved):
Good job!
